lightning/src/ln/channelmanager.rs

```diff
@@ -4567,6 +4686,33 @@ where
 				should_persist = NotifyOption::DoPersist;
 			}
 
+			// Close remaining channels in funding batches when one channel closes.
```

Comment:
This is where the current approach is unclear to me, once we hit `update_maps_on_chan_removal` for one of the funding flows error, we discard all the batch of `(temporary_channel_id, counterparty_node_id)` even if the channel id might belong to N uncorrelated channel counterparties and we have build one funding transaction to save on on-chain fees ? Source of failure might be related to one counterparty only (e.g `peer_disconnected()`) and affect all of them.

Note `funding_transaction_generated` is not very updated on this issue.

Reply:
Yes, that is correct. We need to receive `funding_signed` for all channels to have a local, signed commitment transaction for all funding outputs before we can broadcast. If there is a failure for any channel before receiving `funding_signed`, the remote node will forget the channel and we won't be able to continue.

Or is your point that we might be able to tolerate disconnections for peers that have already signed? It might increase the complexity of the logic though, I'm not sure how large the benefit there is. And the spec mentions the funding node should forget the channel if the funding transaction has not been broadcast yet: https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#requirements-14.